flink-ml-core/src/main/java/org/apache/flink/ml/common/broadcast/operator/AbstractBroadcastWrapperOperator.java

```diff
@@ -169,13 +182,16 @@
                                         parameters.getOperatorEventDispatcher())
                                 .f0;
 
-        boolean hasRichFunction =
+        this.indexOfSubtask = containingTask.getIndexInSubtaskGroup();
+        this.numInputs = inTypes.length;
+
+        this.hasRichFunction =
                 wrappedOperator instanceof AbstractUdfStreamOperator
                         && ((AbstractUdfStreamOperator) wrappedOperator).getUserFunction()
                                 instanceof RichFunction;
 
         if (hasRichFunction) {
-            wrappedOperatorRuntimeContext =
+            this.wrappedOperatorRuntimeContext =
```

Comment:
Is it possible to raise exception if `wrappedOperatorRuntimeContext#getBroadcastVariable` or `wrappedOperatorRuntimeContext#getBroadcastVariableWithInitializer` is invoked on an operator other than the output operator of the `graphBuilder`?

Reply:
It does throws NPE when it is not the output operator of `graphBuilder`. [1]

[1] https://github.com/apache/flink-ml/blob/master/flink-ml-core/src/main/java/org/apache/flink/ml/common/broadcast/BroadcastStreamingRuntimeContext.java#L87